pygount/analysis.py

```diff
@@ -261,10 +262,11 @@ def from_file(
         fallback_encoding: str = "cp1252",
         generated_regexes=pygount.common.regexes_from(DEFAULT_GENERATED_PATTERNS_TEXT),
         duplicate_pool: Optional[DuplicatePool] = None,
+        file_handle: Optional[IOBase] = None,
```

Comment:
Hm, I had some similar requirement in other applications where a file should be processed either from a path or an already open file. What I usually did was to use a single parameter like this:
```
source_path_or_io: Union[str, pathlib.Path, IO]
```
In case of `str` or `pathlib.Path` the function would open and close the file handle by itself. For everything else, it would just assume an open file-like object and just `read()` it. 

However, one issue arising from allowing file like objects in a single parameter is that not all of them have a proper `name` attribute. That prevents pygments from using the name as basis for figuring out the appropriate lexer.

So your approach with 2 parameters makes sense provided the file name can always be obtained and passed along in your own code calling this function.

So in summary, I guess this is fine.

Reply:
Resolved because it is fine.